### ISSUE TITLE
G-Helper freezes when disconnecting XGm and disconnection fails

### ISSUE THREAD

**Issue by Plumbio** [seerge/g-helper#2306]
### Rules

- [X] I made myself familiar with the <a href='https://github.com/seerge/g-helper?tab=readme-ov-file'>Readme</a>, <a href='https://github.com/seerge/g-helper/wiki/FAQ'>FAQ</a> and <a href='https://github.com/seerge/g-helper/wiki/Troubleshooting'>Troubleshooting</a>.
- [X] I understand that, if insufficient information or no app logs will be provided, my issue will be closed without an answer.

### What's wrong?

When I press the XGm button on G-Helper to disconnect it, the window asking if I'm sure appears, but at the same time the AMD Adrenalin Software closes by itself and if y press "yes" on the popped up window, G-Helper freezes and nothing else happens, and I have to wait like 10 seconds for it to unfreeze to try again.
I noticed that if I wait until the AMD software relaunches while the "are you sure?" window is present, I can press "yes" and it will work, so it must be a problem with the Adrenalin Software not being available at that moment.
This problem happens with the the AMD drivers from ASUS's support page and the drivers directly downloaded from AMD.
The AMD Software also closes and relaunches when swtiching from Standard to Eco mode using G-Helper, and it always does, but this didn't happen a while ago and I'm not sure what is causing it, but I have the idea that this started after switching AMD VariBright from G-Helper. I had new AMD drivers cleanly installed and the switch from Standard to Eco was working with no problem, but all of a sudden it just didn't, and the only thing I can recall doing was turning off VariBright from G-Helper.

Just to be clear, the AMD software doesn't show any crash window, the tray icon just disappears and a new one appears like 5 seconds after, and if I have the AMD software window opened at that moment, it will just close.

### How to reproduce the bug?

1. Open G-Helper.
2. Press the XGm button.
3. Quickly confirm your decision on the popped up widnow.

### Logs

[log.txt](https://github.com/seerge/g-helper/files/14767683/log.txt)

### Device and Model

Asus ROG Flow x13 GV301QH

### Armoury Crate

Uninstalled

### Asus Services

0

### Version

0.162.0

### OS

Windows 11 22H2

**Comment by seerge** [comment]
@Plumbio hello, 

Adrenaline closes cause you have checked Stop GPU apps under ``Extra`` :) I don't this is related, but I have added a small delay after killing GPU apps and showing alert

[GHelper.zip](https://github.com/seerge/g-helper/files/14770319/GHelper.zip)

As for UI freezing - this is normal, and unfortunately this is due to how interaction with Asus System Control Interface driver works. Each command locks whole process there. Most of them are instant, so you don't notice it.
Some (like disabling GPU for Eco mode), can take a while ....

**Comment by Plumbio** [comment]
Hi @seerge 
Thanks for your answer, but that shouldn't be the issue cause I don't have "stop gpu apps" checked.
The problem with the XGm not being disconnected after pressing the button is a bit random, it sometimes just works and sometimes it doesn't and g-helper freezes until Adrenalin starts again.

**Comment by seerge** [comment]
@Plumbio ok, can you check if with this build disabling XGM still kills Adrenalin ? 
[GHelper.zip](https://github.com/seerge/g-helper/files/14770785/GHelper.zip)

**Comment by Plumbio** [comment]
Tried a couple of times and Adrenalin didn't get killed, then I restarted my laptop and tried again and Adrenalin survived again so, for now, the problem seems to be fixed.
Adrenalin was still closing when switching to Eco mode before restarting my laptop, but after the restart it is also no longer happening, so even though it was not a real problem, I'm glad it got fixed too.

Thanks a lot @seerge , you are the best.
